### DISCUSSION TITLE
Configuration - Ultralytics YOLOv8 Docs

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#4142] [Docs]
# Configuration - Ultralytics YOLOv8 Docs

Master YOLOv8 settings and hyperparameters for improved model performance. Learn to use YOLO CLI commands, adjust training settings, and optimize YOLO tasks & modes.

https://docs.ultralytics.com/usage/cfg/

**Comment by KristinaTas** [comment]
What does the table under "Augmentation" actually represent?

**Reply by glenn-jocher** [reply]
The Augmentation table shows the hyperparameters available for training augmentations and their default values. For example if you want to randomly rotate all of your training images from -45 to 45 degrees, you can set `degrees=45`, i.e.:

```bash
yolo train model=yolov8n.pt data=coco8.yaml degrees=45
```

**Reply by AhmetEkiz** [reply]
I use `ultralytics==8.0.143` version and train YoloV8 segmentation and object detection models. I have added `degrees` hyperparamaters but training logs doesn't say anything about degree in below line or anywhere. Is it normal? Or how can I check augmentations?

`albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01), CLAHE(p=0.01, clip_limit=(1, 4.0), tile_grid_size=(8, 8))`

**Reply by pderrenger** [reply]
@AhmetEkiz the line you're seeing in the training logs refers to specific augmentations applied using the Albumentations library. The `degrees` hyperparameter you added is part of the YOLOv8 augmentation settings, which may not be explicitly listed in the Albumentations log output.

To verify that the `degrees` augmentation is being applied, you can check your training configuration file (usually a `.yaml` file) to ensure that the `degrees` parameter is set as you intended. If you've set it via the command line or a custom configuration, it should be active during training even if it's not printed in the logs.

The training logs may not detail every augmentation applied, especially if they are part of the default YOLOv8 augmentation pipeline rather than the Albumentations pipeline. If you want to ensure that your augmentations are being applied, you can:

1. Review the configuration file to confirm that the `degrees` parameter is included and set correctly.
2. Enable verbose logging if available, which might provide more detailed information about the augmentations.
3. Visually inspect the augmented images if the framework provides a way to output examples of augmented images during training.

Remember that the augmentations are applied randomly during training, so not every image will be rotated, and the degree of rotation will vary according to the specified range.

**Comment by BhaveshWadibhasme** [comment]
how to add albumentation like augmentation parameters in yolov8

**Reply by pderrenger** [reply]
@BhaveshWadibhasme to add Albumentations-like augmentations in YOLOv8, you can modify the data YAML file associated with your dataset. YOLOv8 supports a variety of augmentation parameters natively, which you can adjust to achieve similar effects to those provided by the Albumentations library.

Here's an example of how you might adjust the augmentation settings in your data YAML file:

```yaml
# Train dataset
train: ../datasets/coco128/images/train2017/
val: ../datasets/coco128/images/val2017/

# Number of classes
nc: 80

# Class names
names: ['person', 'bicycle', ...]

# Augmentation settings
augmentations:
  hsv_h: 0.015  # hue
  hsv_s: 0.7   # saturation
  hsv_v: 0.4   # value
  degrees: 0.0 # rotation
  translate: 0.1 # translation
  scale: 0.5   # scaling
  shear: 0.0   # shearing
  perspective: 0.0 # perspective distortion
  flipud: 0.0  # flip up and down
  fliplr: 0.5  # flip left and right
  mosaic: 1.0  # mosaic
  mixup: 0.0   # mixup
  copy_paste: 0.0 # copy-paste for segmentation
```

Adjust these parameters to match the desired augmentations for your training. Note that while YOLOv8 does not directly integrate Albumentations, the native augmentation capabilities are extensive and can be tuned to achieve a wide range of effects.

For more detailed information on configuring augmentations and other settings, please refer to the [Ultralytics YOLOv8 Documentation](https://docs.ultralytics.com/).

**Comment by BhaveshWadibhasme** [comment]
i wan to to add A.RandomBrightnessContrast(p=0.0),
                A.RandomGamma(p=0.0),
                A.ImageCompression(quality_lower=75, p=0.0) albumentation parameters for training. existing script assigned 0 value to it. but i want to assign some other value to it and use during training.

**Reply by pderrenger** [reply]
@BhaveshWadibhasme to incorporate additional Albumentations parameters like `A.RandomBrightnessContrast`, `A.RandomGamma`, and `A.ImageCompression` with non-zero probabilities into your YOLOv8 training, you'll need to modify the data augmentation settings in your dataset YAML file or directly in your training script if you're using a custom augmentation pipeline.

Here's a concise example of how you might adjust these parameters:

```yaml
# In your dataset.yaml file, under the 'train' section
augmentations:
  RandomBrightnessContrast:
    brightness_limit: 0.2
    contrast_limit: 0.2
    p: 0.5
  RandomGamma:
    gamma_limit: (80, 120)
    p: 0.5
  ImageCompression:
    quality_lower: 75
    quality_upper: 99
    p: 0.5
```

Or, if you're defining the augmentations in your training script:

```python
from ultralytics import YOLO, Augment
import albumentations as A

# Define your custom augmentations
custom_augmentations = Augment(
    A.Compose([
        A.RandomBrightnessContrast(brightness_limit=0.2, contrast_limit=0.2, p=0.5),
        A.RandomGamma(gamma_limit=(80, 120), p=0.5),
        A.ImageCompression(quality_lower=75, quality_upper=99, p=0.5)
    ])
)

# Create a YOLO model and pass the custom augmentations
model = YOLO('yolov8n.yaml', augment=custom_augmentations)

# Train the model
model.train(data='coco128.yaml', epochs=3)
```

Remember to adjust the `p` values to the desired probabilities and set the other parameters to the values that best suit your training needs. For more detailed information on configuring your training, please refer to the [Ultralytics YOLOv8 Documentation](https://docs.ultralytics.com).

**Reply by eljirg** [reply]
I got an error when importing Augment. Which ultralytics version are you using?
I am using ultralytics==8.1.11

**Reply by pderrenger** [reply]
@eljirg hey there! 👋 It looks like you're ready to enhance your YOLOv8 training with some custom Albumentations augmentations. To integrate `A.RandomBrightnessContrast`, `A.RandomGamma`, and `A.ImageCompression` with specific values, you'll need to adjust the augmentation settings in your training configuration.

Here's a quick example of how you might set those up:

```python
from ultralytics import YOLO, Augment
import albumentations as A

# Define your custom augmentations
custom_augmentations = Augment(
    A.Compose([
        A.RandomBrightnessContrast(brightness_limit=0.2, contrast_limit=0.2, p=0.5),
        A.RandomGamma(gamma_limit=(80, 120), p=0.5),
        A.ImageCompression(quality_lower=75, quality_upper=99, p=0.5)
    ])
)

# Create a YOLO model and pass the custom augmentations
model = YOLO('yolov8n.yaml', augment=custom_augmentations)

# Train the model
model.train(data='coco128.yaml', epochs=3)
```

Just tweak the `p` values to set the probabilities for each augmentation, and you're good to go! Happy training! 🚀

**Comment by touma1434** [comment]
if I add 'fliplr=1' will it train on both the original and flipped images?

**Reply by AISTALK** [reply]
probably it will train only on flipped images

**Reply by pderrenger** [reply]
@AISTALK when you set `fliplr=1` in the YOLOv8 configuration, it will indeed apply horizontal flip augmentation to all images during training. This means that each image in your dataset has a 100% chance of being flipped horizontally during the training process. The original images will also be used, so the model effectively sees both the original and flipped versions, enhancing its ability to generalize from the training data. 🔄👍

**Comment by Sencc** [comment]
Hi,yolov8 is a great project.I think it is necessary to explain the use of Augmentation in training.

**Reply by pderrenger** [reply]
@Sencc thank you for your kind words and interest in YOLOv8! 🚀

Augmentations are a crucial part of training deep learning models as they help to introduce variability in the training data, which can improve the model's ability to generalize to new, unseen data. In YOLOv8, you can control various augmentation parameters directly in your training configuration file.

Here are some of the key augmentation parameters you can adjust:

- `hsv_h`, `hsv_s`, `hsv_v`: Adjust the hue, saturation, and value of the image colors to introduce color variability.
- `degrees`: Rotate the image by a certain degree to simulate different orientations.
- `translate`: Translate the image horizontally and vertically to simulate different positions.
- `scale`: Scale the image to simulate different sizes of objects.
- `shear`: Shear the image to simulate perspective changes.
- `perspective`: Adjust the perspective of the image.
- `flipud`, `fliplr`: Flip the image upside down or left to right.
- `mosaic`: Combine four training images into one to simulate different contexts.
- `mixup`: Overlay two images to create a single composite image.
- `copy_paste`: For segmentation tasks, copy objects from one image and paste them onto another.

To use augmentations during training, you can set these parameters in your YAML configuration file or pass them as arguments when initializing the `YOLO` object in Python. For example:

```python
from ultralytics import YOLO

# Initialize YOLOv8 model with custom augmentations
model = YOLO('yolov8n.yaml', imgsz=640, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.1)
```

For more detailed information on each augmentation and how to apply them, please refer to the [Ultralytics YOLOv8 Documentation](https://docs.ultralytics.com/). The documentation provides comprehensive guidance on how to effectively use and customize augmentations for your specific use case. Happy training! 🎯

**Comment by alex19Me** [comment]
I trained on my set of images a model using YOLO8, as a result I saved 'best.pt'. How to write a code in COLAB that will use my results 'best.pt' and will do inference with new set of images?
Thanks for your suppor (I am new to this, but very intrested to learn)

**Reply by AISTALK** [reply]
Just init model as part to best.bt, then inference

**Reply by pderrenger** [reply]
@AISTALK certainly! Here's a concise example of how you can perform inference with your `best.pt` model on a new set of images using Google Colab:

```python
from ultralytics import YOLO

# Load your trained model
model = YOLO('path/to/best.pt')  # Make sure to upload 'best.pt' to Colab or mount your drive

# Perform inference on new images
results = model.predict(source='path/to/your/images')  # Directory containing new images

# Results contain predictions. You can also save or show the images with predictions
results.show()  # Display the images with bounding boxes
results.save()  # Save the images with predictions to the 'runs/detect' folder
```

Make sure to replace `'path/to/best.pt'` with the actual path to your `best.pt` file and `'path/to/your/images'` with the path to your new images. If you're using Google Drive, you can mount it first in Colab and then specify the path accordingly.

Happy coding and best of luck with your YOLOv8 journey! 😊🚀

**Comment by AISTALK** [comment]
Is it possible to use augmentation for certain angle? I want to use angle which are Pi/4*n, n - integer.

**Reply by pderrenger** [reply]
@AISTALK yes, you can customize the augmentation settings in YOLOv8 by adjusting the `degrees` hyperparameter in your training configuration file. The `degrees` hyperparameter controls the rotation augmentation and is specified in degrees.

To rotate images by multiples of π/4 (or 45°), you can set the `degrees` value to 45 in your configuration. This will apply rotation augmentation to your images at angles that are multiples of 45° during training.

Here's an example of how you might set this in your configuration:

```yaml
# Hyperparameters
degrees: 45.0 # Rotate images by multiples of 45 degrees
```

Keep in mind that the actual rotation applied to each image will be a random value from -degrees to +degrees. If you want to strictly rotate by π/4 * n, you would need to implement a custom augmentation function that selects angles from the set {0, 45, 90, 135, ...}.

For more details on configuring your training and augmentation settings, please refer to the [Ultralytics YOLOv8 Documentation](https://docs.ultralytics.com/).

**Comment by AISTALK** [comment]
Hello, how copy_paste actually works? Does it copy a full object to another images or just parts? My object always appears full size, what arguments should I disable to achieve great SOTA performance?

**Reply by pderrenger** [reply]
@AISTALK hello! The `copy_paste` augmentation in YOLOv8 works by copying whole objects from one image and pasting them onto another image. This is done to simulate occlusions and to help the model learn to detect objects even when they are partially obscured.

If you're observing that objects are always pasted in full size and you want to vary this behavior, you might consider adjusting other augmentation parameters that can affect the size and appearance of the pasted objects, such as `scale`, `translate`, `shear`, and `perspective`. However, `copy_paste` itself does not have a direct argument to control the partial copying of objects.

For achieving state-of-the-art (SOTA) performance, it's important to have a balanced combination of augmentations. Disabling `copy_paste` might not necessarily lead to better performance, as it can be beneficial for the model's robustness. Instead, focus on tuning the hyperparameters and using a diverse and representative dataset. You can experiment with different augmentation settings and monitor the validation metrics to guide your adjustments.

Remember to refer to the [Ultralytics Docs](https://docs.ultralytics.com) for detailed information on each parameter and their effects on training. 📘🚀

**Comment by AISTALK** [comment]
"close_mosaic 	10 	(int) disable mosaic augmentation for final epochs (0 to disable)"
0 will disable closing mosaic or mosic augmentation?

**Reply by nishanthballal-9** [reply]
Can anyone confirm on this? How to disable mosaic augmentation?

**Reply by AISTALK** [reply]
you need set this to 0, default value is 1
mosaic 	1.0 	image mosaic (probability)

**Reply by pderrenger** [reply]
@AISTALK to disable mosaic augmentation during training, you should set the `mosaic` hyperparameter to `0`. This will turn off mosaic augmentation entirely. The `close_mosaic` parameter is used to disable mosaic augmentation for the final epochs of training, and setting it to `0` will keep mosaic augmentation enabled for all epochs. 🛠️

**Comment by TakatheUltimate** [comment]
I have downloaded ultralytics per pip as well as all of the requirements, but it always says that yolo is not an internal or external command, I'm a beginner and I have trouble with this as of right now.

**Reply by pderrenger** [reply]
@TakatheUltimate it sounds like you're trying to use YOLOv8 from the command line and encountering an issue where the `yolo` command is not recognized. This typically happens if the package is not installed correctly or if the environment path is not set up to include the location of the installed package.

To resolve this, please ensure that you have activated the correct Python environment where you installed the Ultralytics YOLOv8 package via `pip`. If you're using a virtual environment, you need to activate it first before running any YOLOv8 commands. Here's how you can do it:

1. Open your command line interface (CLI).
2. Activate your Python environment (replace `your_env` with the name of your environment):
   - For Conda: `conda activate your_env`
   - For venv: `source your_env/bin/activate` (on Unix or macOS) or `your_env\Scripts\activate` (on Windows)
3. Once the environment is activated, try running the YOLOv8 commands again.

If you're not using a virtual environment, you might want to check if the `Scripts` folder from your Python installation directory is added to your system's PATH environment variable. This is where the `yolo` command would be located after installation.

If you continue to face issues, you can also use YOLOv8 directly in a Python script without relying on the command line interface. Here's a simple example of how to use YOLOv8 in a Python script:

```python
from ultralytics import YOLO

# Load a pretrained YOLO model
model = YOLO('yolov8n.pt')

# Perform object detection on an image
results = model('path_to_your_image.jpg')

# Results are saved to 'runs/detect/exp' by default
```

Make sure to replace `'path_to_your_image.jpg'` with the actual path to the image you want to process. If you need more detailed guidance, please refer to our documentation for further assistance. 📘

**Comment by AISTALK** [comment]
Does size arg can bee smth like rectangle or it supports only squares?

**Reply by pderrenger** [reply]
@AISTALK the `imgsz` argument in YOLOv8 supports both square and rectangular image sizes. You can specify a single integer to use square images or a list of two integers to use rectangular images. For example:

```python
imgsz=640  # Square image (640x640)
imgsz=[640, 480]  # Rectangular image (640x480)
```

This flexibility allows you to tailor the input size to your specific use case or dataset. 😊👍

**Comment by AISTALK** [comment]
When single_cls might be helpful?

**Reply by pderrenger** [reply]
@AISTALK the `single_cls` setting can be helpful when you're working with a dataset that contains only one class. By setting `single_cls` to `True`, the model will be trained to optimize for single-class detection, which can simplify the output and potentially improve performance since the model won't have to distinguish between multiple classes. This is particularly useful in specialized applications where only one type of object is of interest. 😊👍

**Comment by AISTALK** [comment]
Expose what rect mean please, what does mean it is collated for minimum padding?
rect 	False 	rectangular training with each batch collated for minimum padding

**Reply by AISTALK** [reply]
Is mosaic works for classification?

**Reply by xiaocheng99** [reply]
There is a question about YOLOV8-seg, I have both polygon and rectangle labels at the same time, but in the post-processing part I will hide the mask of the rectangle box label, which will act as a target detection, and then the polygon will continue to do the instance segmentation; therefore, for my case, I hope that in the model output results, for these two labels, respectively, for the evaluation, that is, to put box and mask's related evaluation indexes, for different labels, respectively, to show their related evaluation indexes, which part of the code should I modify and how to do it? I hope to get your answers.

**Reply by xiaocheng99** [reply]
There is a question about YOLOV8-seg, I have both polygon and rectangle labels at the same time, but in the post-processing part I will hide the mask of the rectangle box label, which will act as a target detection, and then the polygon will continue to do the instance segmentation; therefore, for my case, I hope that in the model output results, for these two labels, respectively, for the evaluation, that is, to put box and mask's related evaluation indexes, for different labels, respectively, to show their related evaluation indexes, which part of the code should I modify and how to do it? I hope to get your answers.

**Reply by pderrenger** [reply]
@xiaocheng99 the `rect` parameter in YOLOv8 refers to rectangular training or validation. When set to `True`, it means that during training or validation, the batches of images are collated in a way that minimizes padding. This is achieved by grouping images of similar aspect ratios together, which can lead to more efficient and faster computation since less padding is needed to maintain consistent tensor shapes across a batch.

For example, if you have a batch of images with varying aspect ratios and you do not use rectangular training (`rect=False`), you would need to pad all images to match the dimensions of the largest image in the batch. With rectangular training (`rect=True`), images are grouped by similar aspect ratios, reducing the amount of padding required and potentially speeding up the training process.

Regarding your second question, the mosaic augmentation is typically used for object detection tasks to enhance the model's ability to detect objects at different scales and in various contexts. It is not directly applicable to classification tasks, where the goal is to classify the entire image rather than detecting multiple objects within it.

For your third question about handling both polygon and rectangle labels in YOLOv8-seg, you would need to modify the post-processing code to differentiate between the two label types. You would likely need to adjust the evaluation metrics calculation to handle the two types of labels separately. This could involve customizing the code that processes the model's output and calculates the evaluation indexes for boxes and masks. Since this is a more complex modification, it would be best to refer to the documentation and potentially the source code to understand where and how to implement these changes. If you need further guidance, please refer to the documentation on segmentation tasks and post-processing.

**Comment by xiaocheng99** [comment]
There is a question about YOLOV8-seg, I have both polygon and rectangle labels at the same time, but in the post-processing part I will hide the mask of the rectangle box label, which will act as a target detection, and then the polygon will continue to do the instance segmentation; therefore, for my case, I hope that in the model output results, for these two labels, respectively, for the evaluation, that is, to put box and mask's related evaluation indexes, for different labels, respectively, to show their related evaluation indexes, which part of the code should I modify and how to do it? I hope to get your answers.

**Reply by pderrenger** [reply]
@xiaocheng99 to achieve separate evaluation metrics for rectangle box labels (detection) and polygon labels (instance segmentation) in YOLOv8-seg, you'll need to modify the post-processing and evaluation parts of the code. Here's a high-level overview of the steps you might take:

1. **Modify Dataset Loading**: Ensure your dataset loader differentiates between polygon and rectangle annotations, possibly by adding an attribute to the annotation data structure.

2. **Adjust Model Output**: Modify the model's forward pass to output both bounding boxes and segmentation masks, ensuring that the outputs are tagged with their respective label types.

3. **Customize Post-Processing**: In the post-processing step, separate the outputs based on their label types. Apply the appropriate post-processing for detection (NMS for boxes) and segmentation (mask thresholding).

4. **Separate Evaluation Metrics**: Implement or modify the evaluation metrics to handle the two types of outputs separately. You might need to write custom functions or modify existing ones to calculate metrics like mAP for detection and IoU for segmentation masks.

5. **Results Handling**: Ensure that the results are stored and displayed separately, with clear distinctions between the metrics for detection and segmentation.

You'll likely need to dive into the codebase and make changes in the parts of the code responsible for dataset handling, model forward pass, post-processing, and evaluation. Since this is a non-trivial task, it would require a good understanding of the codebase and the underlying logic of YOLOv8-seg.

For detailed guidance on the specific code changes, please refer to the [Ultralytics Docs](https://docs.ultralytics.com), particularly the sections on custom datasets, model customization, and evaluation. If you encounter any issues or need further assistance, feel free to open an issue on the GitHub repository with details of your implementation, and the community will be happy to help you out. 🛠️🧠

**Comment by xiaocheng99** [comment]
There is a question about YOLOV8-seg, I have both polygon and rectangle labels at the same time, but in the post-processing part I will hide the mask of the rectangle box label, which will act as a target detection, and then the polygon will continue to do the instance segmentation; therefore, for my case, I hope that in the model output results, for these two labels, respectively, for the evaluation, that is, to put box and mask's related evaluation indexes, for different labels, respectively, to show their related evaluation indexes, which part of the code should I modify and how to do it? I hope to get your answers.

**Reply by pderrenger** [reply]
@xiaocheng99 for your use case with YOLOv8-seg, where you have both polygon and rectangle labels and want to evaluate detection and segmentation separately, you'll need to modify the post-processing and evaluation code to handle these two types of annotations differently.

Here's a high-level overview of the steps you might take:

1. **Modify Dataset Loading**: Ensure your dataset loader distinguishes between polygon and rectangle annotations, possibly by adding an attribute to the annotation data structure.

2. **Adjust Model Output**: If necessary, adjust the model's output to include separate predictions for boxes and masks, ensuring that the model can output both even if they originate from the same object class.

3. **Post-Processing**: In the post-processing step, separate the predictions into two groups based on the type of label (polygon or rectangle). Apply the respective evaluation metrics to each group.

4. **Evaluation Metrics**: Implement or modify the evaluation metrics to handle the two types of predictions. For detection, you might use metrics like Precision, Recall, and mAP for the bounding boxes. For segmentation, you might use metrics like IoU and mIoU for the masks.

5. **Results Presentation**: Finally, modify the code that presents the evaluation results to display the metrics for boxes and masks separately.

You'll likely need to dive into the code that handles dataset loading, model output, post-processing, and evaluation. Look for the parts of the code where predictions are processed and where evaluation metrics are calculated. You may need to create custom functions or modify existing ones to achieve the desired separation of metrics.

Remember to test your modifications thoroughly to ensure that the separate evaluations for detection and segmentation are accurate and that no unintended changes have been introduced to the model's behavior.

For detailed guidance on modifying specific parts of the code, please refer to the documentation and the codebase itself, as direct code modification advice is beyond the scope of this platform. If you encounter specific issues or need further clarification, feel free to open an issue in the repository with a detailed description of your problem. Good luck with your project! 🚀

**Comment by cpanaintescu** [comment]
how can I create a region of interest?

**Reply by pderrenger** [reply]
@cpanaintescu to create a region of interest (ROI) with YOLOv8, you would typically preprocess your images or videos before passing them to the model for prediction. Here's a concise way to define an ROI within an image using Python:

```python
import cv2
from ultralytics import YOLO

# Load your model
model = YOLO('yolov8n.pt')

# Read your image
image = cv2.imread('path/to/image.jpg')

# Define your ROI coordinates (x1, y1, x2, y2)
x1, y1, x2, y2 = 100, 100, 400, 400

# Crop the image to the ROI
roi = image[y1:y2, x1:x2]

# Perform prediction on the ROI
results = model.predict(source=roi)

# Process results as needed
for result in results:
    # Handle detection results within the ROI
    pass
```

Remember to adjust the ROI coordinates (`x1, y1, x2, y2`) to the specific area you're interested in. The `predict` method will then perform object detection within this cropped region. If you need to map the detected objects' coordinates back to the original image space, you'll need to offset them by `(x1, y1)`.

**Comment by MilenioScience** [comment]
Hello, how to applied augmentation for training custom dataset yolov8?

**Reply by pderrenger** [reply]
@MilenioScience to apply augmentations during the training of a custom dataset with YOLOv8, you can modify the data YAML file associated with your dataset. Here's a concise guide on how to do it:

1. Open your custom dataset's YAML file (e.g., `custom_dataset.yaml`).
2. Add or modify the augmentation parameters under the `train:` section. For example:

```yaml
    hsv_h: 0.015  # hue augmentation
    hsv_s: 0.7    # saturation augmentation
    hsv_v: 0.4    # value augmentation
    degrees: 0.0  # rotation
    translate: 0.1  # translation
    scale: 0.5    # scaling
    shear: 0.0    # shearing
    perspective: 0.0  # perspective distortion
    flipud: 0.0   # vertical flip
    fliplr: 0.5   # horizontal flip
    mosaic: 1.0   # mosaic augmentation
    mixup: 0.0    # mixup augmentation
    copy_paste: 0.0  # copy-paste augmentation
```

3. Save the changes to your YAML file.
4. Use this YAML file when initiating training with the `model.train()` function.

For more detailed information on each augmentation and how to configure them, please refer to the [Ultralytics YOLOv8 Documentation](https://docs.ultralytics.com/). 📘🚀
